### PR TITLE
Remove the old docker-multinode files that were built into the hyperkube image

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

ref: https://goo.gl/VxSaKx

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
The hyperkube image has been slimmed down and no longer includes addon manifests and other various scripts. These were introduced for the now removed docker-multinode setup system.
```

cc @jbeda @brendandburns @bgrant0607 @justinsb @mikedanese

### CODE REVIEW

**Comment by roberthbailey** [comment]
lgtm (not adding the slash to get relevant approvals before this gets merged)

**Comment by mikedanese** [comment]
lgtm as well

/approve

**Comment by luxas** [comment]
Cool, thanks! @jbeda is the last one to lgtm then before merge :+1:

**Comment by jbeda** [comment]
This LGTM but is a bit scary.

I'd put a bit more in the release note.  Something like "The hyperkube image has been slimmed down and no longer includes addon manifests and other various scripts.  These were introduced for the now removed docker-multinode setup system."

/lgtm

**Comment by jbeda** [comment]
Sorry I didn't get this earlier!  I thought I had covered all of these.

**Comment by luxas** [comment]
> This LGTM but is a bit scary.

Yes, these were here only for the docker-multinode setup

> The hyperkube image has been slimmed down and no longer includes addon manifests and other various scripts. These were introduced for the now removed docker-multinode setup system.

Proposal accepted ;)

Thanks!

**Comment by nikhiljindal** [comment]
I believe this PR broke federation build.

I see the following in logs:

```
I0420 10:48:18.314] make[1]: Entering directory '/var/lib/jenkins/workspace/ci-kubernetes-federation-build/go/src/k8s.io/kubernetes/cluster/images/hyperkube'
I0420 10:48:18.719] tar: /tmp/hyperkubeD9ftu6/cni-bin: Cannot open: No such file or directory
I0420 10:48:18.719] tar: Error is not recoverable: exiting now
I0420 10:48:18.720] curl: (23) Failed writing body (919 != 1379)
I0420 10:48:18.721] Makefile:62: recipe for target 'build' failed
I0420 10:48:18.721] make[1]: *** [build] Error 2
I0420 10:48:18.722] make[1]: Leaving directory '/var/lib/jenkins/workspace/ci-kubernetes-federation-build/go/src/k8s.io/kubernetes/cluster/images/hyperkube'
I0420 10:48:18.725] !!! [0420 10:48:18] Call tree:
I0420 10:48:18.727] !!! [0420 10:48:18]  1: ./develop/develop.sh:151 build_image(...)
I0420 10:48:18.727] !!! Error in ./develop/develop.sh:112
I0420 10:48:18.727]   Error in ./develop/develop.sh:112. '((i<3-1))' exited with status 2
I0420 10:48:18.727] Call stack:
I0420 10:48:18.727]   1: ./develop/develop.sh:112 build_image(...)
I0420 10:48:18.727]   2: ./develop/develop.sh:151 main(...)
I0420 10:48:18.727] Exiting with status 1
I0420 10:48:18.728] Makefile:50: recipe for target 'build_image' failed
```

Was removing `cni-bin` dir intentional?

File: cluster/images/hyperkube/Makefile [hunk 1]
```diff
@@ -60,43 +60,15 @@ ifndef VERSION
     $(error VERSION is undefined)
 endif
 	cp -r ./* ${TEMP_DIR}
-	mkdir -p ${TEMP_DIR}/cni-bin ${TEMP_DIR}/addons ${TEMP_DIR}/addons/singlenode ${TEMP_DIR}/addons/multinode
```
Comment (nikhiljindal) [inline cluster/images/hyperkube/Makefile]: Looks like there are other scripts relying on this `cni-bin` dir. Was removing this intentional?
It seems to have broken federation build.

**Comment by nikhiljindal** [comment]
Sent https://github.com/kubernetes/kubernetes/pull/44751 to fix the issue. Looking for someone in `cluster/OWNERS` to approve.

**Pull request merged**